task7/script.js

```diff
@@ -0,0 +1,57 @@
+const documentFragment = document.createDocumentFragment();
+
+const ul = document.getElementById("ul");
+
+const listItem = [{
+  to: "bookmark.html",
+  img: "1.png",
+  alt: "画像1",
+  text: "ブックマーク"
+}, {
+  to: "message.html",
+  img: "2.png",
+  alt: "画像2",
+  text: "メッセージ"
+}];
+
+const getListItem = () => {
+  return new Promise((resolve) => {
+    resolve(listItem);
+    insertLoading();
+  });
+};
+
+// ローディング画像を表示
+const insertLoading = () => {
+  const loading = document.getElementById("loading");
+  const img = document.createElement("img");
+  img.src = "loading-circle.gif";
+  img.id = "loadingImg";
+  loading.appendChild(img);
+}
+
+getListItem().then((value) => {
+  setTimeout(() => {
+    // ローディング画像を非表示
+    const img = document.getElementById("loadingImg");
+    img.src = "";
+    insertUl(value);
+  }, 3000);
+});
+
```

Comment:
So we do not need to use 'setTimeout' here. This meaning just delete the image and show insertUl after 3 seconds. (it is not meaning resolved)

Reply:
This has been deleted due to the previous commit.